Tests/FuzzilliTests/CompilerTests/basic_delete.js

```diff
@@ -0,0 +1,22 @@
+if (typeof output === 'undefined') output = console.log;
+
+const obj = { a: 1 };
+console.log(delete obj.a);
+console.log(obj);
+
+const propName = 'b';
+obj[propName] = 2;
+console.log(delete obj[propName]);
+console.log(obj);
+
+const arr = [1, 2, 3];
+console.log(delete arr[1]);
+console.log(arr);
+
+const index = 0;
+console.log(delete arr[index]);
+console.log(arr);
+
+const nestedObj = { a: { b: 2 } };
+console.log(delete nestedObj?.a?.b);
```

Comment:
Maybe also add a test for something like
```
try {
    delete null.a;
} catch(e) {
    console.log(e.message);
}
console.log(delete null?.a);
```

Reply:
This is compilable. However, this would cause DeleteProperty to be called, even though null is not an object. In Compiler.swift, I believe it is currently not possible to check [here](https://github.com/TobiasWienand/fuzzilli/blob/530e5099881982589be546fa4c00af7c5a6e786f/Sources/Fuzzilli/Compiler/Compiler.swift#L994) if object is truly an object. We could either
- Try preventing this in parser.js, although there are possibly more edge cases than just this
- Ignore it and add a TODO
- Introduce a new DeleteAny JSOperation

What would you prefer?

Reply:
No I think this is all working exactly as it should :) We  We should just assume that the LHS of the MemberExpression for `delete` is an object.